src/audio/dcblock/Kconfig

```diff
@@ -6,3 +6,24 @@ config COMP_DCBLOCK
 	help
 	  Select for DC Blocking Filter component. This component filters out
 	  the DC offset which often originates from a microphone's output.
+
+choice
+	prompt "DC Blocking Filter HIFI level"
+	depends on COMP_DCBLOCK
+
+# The first valid one seems to be the default.
+# If not reliably, this can also be achieved with 3
+# explicit `default` lines above.
```

Comment:
ok, so does ```XTENSA_ARCH_HAS_HIFI3``` etc come from Zephyr Kconfig ? I guess some of this is still WIP on the Zephyr side.

Reply:
> ok, so does XTENSA_ARCH_HAS_HIFI3 etc come from Zephyr Kconfig ? I guess some of this is still WIP on the Zephyr side.

Yes, as mentioned above it's https://github.com/zephyrproject-rtos/zephyr/pull/67513. It didn't draw huge crowds but the little feedback has been positive so far.

The main problem is: how to duplicate 67513 in XTOS and test it? XTOS was removed from the main branch for Intel. Maybe I can try to use a Cadence/Intel toolchain for compile-testing some non-Intel platform? Either way this will duplicate a lot of macros which are already duplicating `core-isa.h`. Much smaller alternative #8720 keeps relying on `core-isa.h` which means it has none of these problems.

Reply:
67513 just provide option for sof build, we can further define specifc HIFI version in each platform config.
like mtlp, https://github.com/thesofproject/sof/blob/main/app/boards/intel_adsp_ace15_mtpm.conf
add XTENSA_ARCH_HAS_HIFI4 to it, then as long as you are buiding this platform, all version have HIFI 4 will be selected.

I realized there is another problem, some module have HIFI4 and HIFI3 both, however, we can only select one, this is the difficult part, seems we are still not get this done even with zephyr side define.